lib/resque/worker.rb

```diff
@@ -224,11 +224,13 @@ def work(interval = 5.0, &block)
       loop do
         break if shutdown?
 
-        unless work_one_job(&block)
+        if work_one_job(&block)
+          @signal_handler.handle_signal(0)
+        else
           break if interval.zero?
```

Comment:
this line here might mess up your logic.. if the interval is zero and the queue is empty, then we wont handle signals...